api/openapi-spec/swagger.json

```diff
@@ -999,7 +1003,11 @@
           "description": "Rolling update config params. Present only if DeploymentStrategyType = RollingUpdate."
         },
         "type": {
-          "description": "Type of deployment. Can be \"Recreate\" or \"RollingUpdate\". Default is RollingUpdate.",
+          "description": "Type of deployment. Can be \"Recreate\" or \"RollingUpdate\". Default is RollingUpdate.\n\nPossible enum values:\n - `Recreate`: Kill all existing pods before creating new ones.\n - `RollingUpdate`: Replace the old ReplicaSets by new one using rolling update i.e gradually scale down the old ReplicaSets and scale up the new one.",
```

Comment:
This feels nit-picky, but the quoting is all over the map here.  Would it be possible (or maybe a bad idea) to consistently "double quote" string values?

Reply:
The quotes are the results of Marshaling to JSON. User-faced interfaces (generated OpenAPI docs, UIs, etc) will display the quoted strings as-is.

Reply:
I don't understand - json doesn't use back-quotes at all.  I'd expect this to look like:

```
"description": "Type of deployment. Can be \"Recreate\" or \"RollingUpdate\". Default is \"RollingUpdate\".\n\nPossible enum values:\n - \"Recreate\": Kill all existing pods before creating new ones.\n - \"RollingUpdate\": Replace the old ReplicaSets by new one using rolling update i.e gradually scale down the old ReplicaSets and scale up the new one.",
```

Reply:
Sorry for the confusion. This value comes from https://github.com/kubernetes/kubernetes/blob/v1.23.0-alpha.4/pkg/apis/apps/types.go#L377

I guess this is part of the comment convention issue.

Reply:
That explains the first part.  The back-quotes on the enum-values in the added comment come from SOMEWHERE.

Reply:
The back-quotes are intended. OpenAPI specification defines all `Description` as text or Markdown format (or CommonMark to be exact) [1] . 
These back-quoted values rendered great in markdown.

Type of deployment. Can be "Recreate" or "RollingUpdate". Default is RollingUpdate.

Possible enum values:
- `Recreate`: Kill all existing pods before creating new ones.
- `RollingUpdate`: Replace the old ReplicaSets by new one using rolling update i.e gradually scale down the old ReplicaSets and scale up the new one.


[1] https://swagger.io/specification/#rich-text-formatting

Reply:
I find 2 things weird:

1) In the same comment we put this value in double-quotes, back-quotes, and no quotes.  We should pick one.
2) We are not quoting string values.  I know YAML doesn't strictly care (most of the time) but sometimes it does.  Consider places where we use "True" as a string.  If we do not quote that, YAML turns it into a bool.

The argument I am making is that we should  set the convention that string values get quotes.  If Description is expected to be markdown (WEIRD!) then we  should decide to either use back-quotes AND double-quotes or just double-quotes.

E.g.

```
"description": "Type of deployment. Can be `\"Recreate\"` or `\"RollingUpdate\"`. Default is `\"RollingUpdate\"`.\n\nPossible enum values:\n - `\"Recreate\"`: Kill all existing pods before creating new ones.\n - `\"RollingUpdate\"`: Replace the old ReplicaSets by new one using rolling update i.e gradually scale down the old ReplicaSets and scale up the new one.",
```

I know I am being a pain, so I am willing to back off at some point.  My concern is that it is ambiguous.

Reply:
(1) That's true. During revisiting the comment generator, I will have a way to detect and correct quotes in the input. For example, replace all "quotes" with one back-quote. Note: input means the comments in Go source code. The generator reads Go source file and produce a generated Spec.
(2) Fortunately `swagger.json` is a JSON and thus values always have a outer-quote. I don't see we use the raw value of `Description` as part of YAML, so I don't think this could cause an issue.

Reply:
The question is whether, when a user looks at the docs, they see a probably-string-but-open-to-interpretation or an unambiguous string.

If we are going to make a pass through the string, it would be nice to just assert that all instances of the value-in-question are either back-quoted double-quoted strings or just double-quoted strings.  I'll make this my last comment - I think you get the point, right? :)

Reply:
Got the point. I will update the description generation during alpha-to-beta graduation.

Reply:
We have time to fix this before code-freeze. Can you please update the PR so that the string is consistent?

Reply:
I was about to type exactly what Tim wrote above:
```
"description": "Type of deployment. Can be `\"Recreate\"` or `\"RollingUpdate\"`. Default is `\"RollingUpdate\"`.\n\nPossible enum values:\n - `\"Recreate\"`: Kill all existing pods before creating new ones.\n - `\"RollingUpdate\"`: Replace the old ReplicaSets by new one using rolling update i.e gradually scale down the old ReplicaSets and scale up the new one.",
```

Reply:
Which renders  as:
----------------------------------

"description": "Type of deployment. Can be `"Recreate"` or `"RollingUpdate"`. Default is `"RollingUpdate"`.

Possible enum values:
 - `"Recreate"`: Kill all existing pods before creating new ones.
 - `"RollingUpdate"`: Replace the old ReplicaSets by new one using rolling update i.e gradually scale down the old ReplicaSets and scale up the new one.

Reply:
Yes, that's perfect.

Reply:
Updated with in the `k/kube-openapi`, and update dependencies here.

Reply:
Re-running CIs

Reply:
Sorry for the confusion and all the back-and-fourth. I've sent an issue to track the progress of polishing our comments across all types to make quotes consistent. https://github.com/kubernetes/kubernetes/issues/106437

This is a problem of our docs, not a single doc or spec generator like this one.